### PR TITLE
docs: Update Getting Started guide.

### PR DESCRIPTION
**Pull request by morsapaes**
This PR is a proposal to update the Getting Started guide to (hopefully) make it quicker and more straightforward to follow. 

The easiest way to review it would be to build the docs locally and have a look! Here's a rough changelog:

### Get Started

- Included installation steps in the guide (see [Tabs](#tabs)).
- Proposed a new getting started flow based on the existing guide for Cloud (using PubNub).
- Added a "What's Next?" section to round things off and highlight a couple of additional examples like joins and temporal filters (see [Expandable Sections](#expandable-sections)).

### Install

- Added missing instructions to install `psql` on macOS and Linux. This addresses issue #6719 .

## Shortcodes

> 😅 I'm new to Hugo and historically not a CSS pro, so if anyone feels like beautifying these — go ahead!

### Tabs
To have the installation instructions on the Getting Started page instead of sending users to a different page, I added some shortcodes for tabs. It looks a bit weird in combination with these numbered sections, but I couldn't think of a way to make it look less bad.
Screenshot 2021-07-22 at 13 39 32

### ~Expandable sections~ (Removed)
~As an optional extension to the guide, I wanted to include some common patterns for users to try (also to highlight nice Materialize features off the bat). This would make the page quite overwhelming, so my approach was to add a shortcode for expandable sections:~

<hr>

**TODO**
Once there's some feedback on the revamped guide and things are looking final:
- [ ] Include the new shortcodes in the style guide.
- [ ] (Separate PR) Adapt the Cloud Getting Started guide.
- [ ] (Separate PR) Adapt Katacoda (if we want to keep this, @awang ?)

### CODE REVIEW

**Comment by morsapaes** [comment]
Ugh, though I'd checked all the links. Will take care of the failing checks!

**Comment by ruf-io** [comment]
This is a huge improvement.

I think if Joins and Temporal Filters are important enough to include in the quick start, they shouldn't be hidden behind toggles. My rationale:

- some people won't expand the toggles
- we're just delaying any abandonment due to overwhelmingness until after they expand the toggles.
- I think we can present it in a way where Joins and Temporal Filter content is shown by default, but doesn't feel as overwhelming.

**Idea for alternative to toggles:**
Can we come up with toy examples of Joins and Temporal Filters that fit into the body of the getting started guide and can be completed in a reasonable amount of time, and link off to the more detailed existing pages?

**Comment by msudberg** [comment]
This looks really great! 

What do you think about toggling the installation instructions instead of using tabs? That would look more consistent. We could also make the toggles:

* To use a Materialize Cloud account instead of the installed version, click here.
* To install with Docker
* To install with HomeBrew on MacOS
* To install with apt on Linux

and cutting the "Assuming that you're using" wording.

I agree with @ruf-io about not toggling joins and temporal filters. Instead of "What's next?", let's put them under h2s for "Joins" and "Time-windowed computation".

File: doc/user/content/get-started.md [hunk 1]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
 ---
 
-To help you get started with Materialize, we'll:
+This guide walks you through getting started with Materialize, from installing it to creating your first materialized view on top of streaming data. We'll cover:
 
-- Install, run, and connect to Materialize
-- Explore its API
-- Set up a real-time stream to perform aggregations on Wikipedia user edits.
+* Installing, running and connecting to Materialize
+
```
Comment (msudberg) [inline doc/user/content/get-started.md:11]: Serial comma after running :)

File: doc/user/content/get-started.md [hunk 2]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+* Connecting to a streaming data source
+
+* Creating a materialized view
+
+* Exploring common patterns like joins and time-windowing
+
```
Comment (msudberg) [inline doc/user/content/get-started.md:17]: Let's break this out into two bullets:

* Trying out joins
* Exploring time-windowed computations, such as "Updates from the last ten minutes" or "Orders from the last day, broken out by hour".

File: doc/user/content/get-started.md [hunk 3]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+{{< tab "macOS">}}
 
+1. Assuming you're using [Homebrew](https://brew.sh/), open a terminal and run:
+    
     ```shell
-    psql -U materialize -h localhost -p 6875 materialize
+    brew install MaterializeInc/materialize/materialized
     ```
+    
+    **Note:** For a _curl_-based alternative, see [Install](/install/#curl).
```
Comment (msudberg) [inline doc/user/content/get-started.md:56]: "curl" should be code, not italics.

For all of the psql commands, can you specify that the user should open another terminal to run them? That may be too basic for our user base, lot of our new employees haven't realized this until told.

File: doc/user/content/get-started.md [hunk 4]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+    Otherwise, you can find the steps to install and use your CLI of choice under [Install](/install/#cli-connections).
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+## Explore a streaming source
+
+Materialize allows you to work with streaming data from multiple external sources using nothing but standard SQL. You write arbitrarily complex queries, Materialize takes care of maintaining the results automatically up to date with very low latency.
+
```
Comment (msudberg) [inline doc/user/content/get-started.md:115]: The comma after "arbitrarily complex queries" should be a semicolon. (I know it was already like that.)

File: doc/user/content/get-started.md [hunk 5]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+    CREATE SOURCE market_orders_raw 
+    FROM PUBNUB
+    SUBSCRIBE KEY 'sub-c-4377ab04-f100-11e3-bffd-02ee2ddab7fe'
+    CHANNEL 'pubnub-market-orders';
     ```
 
-1. Determine the sum of all keys' sums:
+    The `CREATE SOURCE` statement is a definition of where to find and how to connect to our data source — Materialize won't start ingesting data just yet.
+
+    To list the columns created:
```
Comment (msudberg) [inline doc/user/content/get-started.md:129]: I would break the column listing into its step.

File: doc/user/content/get-started.md [hunk 6]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
     ```
 
-    We can actually then save this query as its own materialized view:
+    Otherwise, you can find the steps to install and use your CLI of choice under [Install](/install/#cli-connections).
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+## Explore a streaming source
```
Comment (msudberg) [inline doc/user/content/get-started.md:112]: ## Stream data

File: doc/user/content/get-started.md [hunk 7]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+           AVG(bid_price) AS avg
+    FROM market_orders
+    GROUP BY symbol;
     ```
 
-Of course, these are trivial examples, but hopefully begin to illustrate some of Materialize's potential.
+    The `avg_bid` view is incrementally updated as new data streams in, so you get fresh and correct results with millisecond latency. Behind the scenes, Materialize is indexing the results of the embedded query in memory (i.e. _materializing_ the view).
 
-## Create a real-time stream
+    To see the results:
```
Comment (msudberg) [inline doc/user/content/get-started.md:163]: Make this its own step.

File: doc/user/content/get-started.md [hunk 8]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
-            (val->'timestamp')::float AS r_ts,
-            val->>'title' AS title,
-            val->>'type' AS type,
-            val->>'user' AS user,
-            val->>'wiki' AS wiki
-        FROM (SELECT data::jsonb AS val FROM wikirecent);
-    ```
-
-1. From here we can start building our aggregations. The simplest place to start is simply counting the number of items we've seen:
+Joins are one of the most powerful features in SQL, but something quite hard to get right in the streaming world. Materialize efficiently supports [all types of SQL joins](/sql/join/#examples) under all the conditions you would expect from a traditional relational database.
```
Comment (msudberg) [inline doc/user/content/get-started.md:195]: ## Joins

File: doc/user/content/get-started.md [hunk 9]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+
+1. Create and populate a table with static, reference data:
```
Comment (msudberg) [inline doc/user/content/get-started.md:198]: No comma after "static".

File: doc/user/content/get-started.md [hunk 10]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+                 ('Google','GOOG'), 
+                 ('Elerium','ELER'), 
+                 ('Bespin Gas','BGAS'),
+                 ('Linen Cloth','LCLO')
+    );
+
     ```
 
-1. However,  we can also see more interesting things from our stream. For instance, who are making the most changes to Wikipedia?
+    **Note:** We are using a table for convenience, to avoid adding complexity to the guide. It's [unlikely](/sql/create-table/#when-to-use-a-table) that you'll need to use tables in real-world scenarios.
```
Comment (msudberg) [inline doc/user/content/get-started.md:216]: Let's put this in the new note formatting. Also doesn't need a comma after "convenience".

File: doc/user/content/get-started.md [hunk 11]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+
+1. Now, we can enrich our aggregated data with the ticker for each stock using a regular `JOIN`:
```
Comment (msudberg) [inline doc/user/content/get-started.md:218]: No comma after "Now".

File: doc/user/content/get-started.md [hunk 12]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
-        SELECT user, count(*) FROM recentchanges GROUP BY user;
+    CREATE MATERIALIZED VIEW cnt_ticker AS
+    SELECT s.ticker AS ticker, 
+           COUNT(*) AS cnt
+    FROM market_orders m
+    JOIN symbols s ON m.symbol = s.symbol
+    GROUP BY s.ticker;
     ```
 
+    To see the results:
```
Comment (msudberg) [inline doc/user/content/get-started.md:229]: Make its own step.

File: doc/user/content/get-started.md [hunk 13]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+     BGAS   |  49
+     ELER   |  68
+     GOOG   |  51
+     LCLO   |  70
     ```
+{{< /expand >}}
 
-1. If this is a factoid we often want to know, we could also create a view of just the top 10 editors we've seen.
+{{< expand "Temporal Filters" >}}
+
```
Comment (msudberg) [inline doc/user/content/get-started.md:244]: ## Temporal filters

File: doc/user/content/get-started.md [hunk 14]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+A common requirement in streaming use cases is time-windowing, for example to model business processes or simply to bound resource usage. In Materialize, you can express this pattern using [temporal filters](/guides/temporal-filters/).
```
Comment (msudberg) [inline doc/user/content/get-started.md:245]: Comma after "for example".

File: doc/user/content/get-started.md [hunk 15]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+
+1. If instead of computing and maintaining the _overall_ count we want to get the _moving_ count over the past minute:
```
Comment (msudberg) [inline doc/user/content/get-started.md:248]: Commas after "If" and the first "count", although actually I think we can cut the phrase altogether. We should probably be a little more descriptive in the step:

"To get the count of stock orders made over the last minute, instead of setting a static time, we create a materialized view using a temporal filter that will always compute results from *now*, whenever *now" happens to be:

File: doc/user/content/get-started.md (hunk 14 above)
Comment (msudberg) [inline doc/user/content/get-started.md:245]: I think we need to define "time-windowing" again here.

File: doc/user/content/get-started.md [hunk 16]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
-            val->>'type' AS type,
-            val->>'user' AS user,
-            val->>'wiki' AS wiki
-        FROM (SELECT data::jsonb AS val FROM wikirecent);
-    ```
-
-1. From here we can start building our aggregations. The simplest place to start is simply counting the number of items we've seen:
+Joins are one of the most powerful features in SQL, but something quite hard to get right in the streaming world. Materialize efficiently supports [all types of SQL joins](/sql/join/#examples) under all the conditions you would expect from a traditional relational database.
+
+1. Create and populate a table with static, reference data:
```
Comment (msudberg) [inline doc/user/content/get-started.md:197]: Do we need to create a table here? Can't we use the PubNub stream?

File: doc/user/content/get-started.md [hunk 17]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+{{< expand "Temporal Filters" >}}
+
+A common requirement in streaming use cases is time-windowing, for example to model business processes or simply to bound resource usage. In Materialize, you can express this pattern using [temporal filters](/guides/temporal-filters/).
+
+1. If instead of computing and maintaining the _overall_ count we want to get the _moving_ count over the past minute:
 
     ```sql
-    CREATE MATERIALIZED VIEW top10 AS
-        SELECT * FROM useredits ORDER BY count DESC LIMIT 10;
+    CREATE MATERIALIZED VIEW cnt_sliding AS
```
Comment (msudberg) [inline doc/user/content/get-started.md:250]: If you're running this on a static table, it won't make work unless you've made an update in the past minute. It makes more sense to run it on a PubNub stream, I think.

File: doc/user/content/get-started.md [hunk 18]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
+     Bespin Gas  |  35
+     Linen Cloth |  45
     ```
 
-Naturally, there are many interesting views of this data. If you're interested
-in continuing to explore it, you can check out the stream's documentation from
-Wikipedia.
-
-Once you're done, don't forget to stop `curl` and `rm wikirecent`.
+{{< /expand >}}
```
Comment (msudberg) [inline doc/user/content/get-started.md:272]: For temporal filters, we should tell people to wait a few minutes and retrieve results again, so they can see the results changing over time (the sliding of the window).

File: doc/user/content/get-started.md [hunk 19]
```diff
@@ -5,216 +5,271 @@ menu: "main"
...
 
 {{< cta href="/demos/microservice" >}}
 Next, see how Materialize can work as an entire microservices →
```
Comment (msudberg) [inline doc/user/content/get-started.md:275]: I don't think "microservices" can be used as singular. Maybe "microservices platform"? There should be a period at the end.

File: doc/user/content/install.md [hunk 20]
```diff
@@ -128,7 +128,36 @@ Detail | Info
 **Database** | `materialize`
 **Port** | `6875`
 
-For more information, see [CLI Connections](/connect/cli/).
+### CLI Connections
+
+To connect to a running instance, you can use any [Materialize-compatible CLI](connect/cli/), like `psql` or `mzcli`. To install e.g. the `psql` client:
+
```
Comment (msudberg) [inline doc/user/content/install.md:134]: Remove "e.g."

File: doc/user/content/install.md [hunk 21]
```diff
@@ -128,7 +128,36 @@ Detail | Info
...
+Once you have `psql` installed, connect using:
+
+```shell
+psql -U materialize -h localhost -p 6875 materialize
+```
+
+<p>
 
 {{< cta href="/get-started" >}}
 Next, let's get started with Materialize →
```
Comment (msudberg) [inline doc/user/content/install.md:163]: End with period.

**Review by msudberg — CHANGES_REQUESTED** [review]

File: doc/user/content/get-started.md (hunk 5 above)
Comment (morsapaes) [reply doc/user/content/get-started.md:129]: Since I'm not really going into the output of this statement, it's more of a "execute if you're curious" than a real step.

File: doc/user/content/get-started.md (hunk 15 above)
Comment (morsapaes) [reply doc/user/content/get-started.md:248]: You're absolutely right, this could be a bit more helpful. I wanted to avoid going into the whole explaining `mz_logical_timestamp()`, but it does lack context as is.

File: doc/user/content/get-started.md (hunk 16 above)
Comment (morsapaes) [reply doc/user/content/get-started.md:197]: There isn't really anything to join with there, unless you do something weird like a self-join.

Comment (morsapaes) [reply doc/user/content/get-started.md:197]: That os the main thing with showing a join, you need some sort of reference data. We could potentially define a set as a sub-query, but this could be confusing.

File: doc/user/content/get-started.md (hunk 17 above)
Comment (morsapaes) [reply doc/user/content/get-started.md:250]: Not sure I get your feedback, this is running on the `market_orders` view which is based on the PubNub source.

File: doc/user/content/get-started.md (hunk 18 above)
Comment (morsapaes) [reply doc/user/content/get-started.md:272]: I'll add some more context, it does end quite abruptly!

File: doc/user/content/get-started.md (hunk 19 above)
Comment (morsapaes) [reply doc/user/content/get-started.md:275]: This was already in there, so I just kept it. I don't think this is the right follow-up to the getting started guide, as it's way too complex. But this is a general problem we have with demos, so happy to discuss what we could use here instead. Maybe the BI Demo would be a softer transition, while we don't have more accessible demos at hand, @ruf-io @msudberg ?

File: doc/user/content/get-started.md (hunk 6 above)
Comment (morsapaes) [reply doc/user/content/get-started.md:112]: I might be biased here, but the main purpose of Materialize is not to stream data, but to consume and transform streams of data. So "Stream data" wouldn't be correct here, IMO.

**Comment by morsapaes** [comment]
Thanks for the feedback, @msudberg and @ruf-io ! Addressing each of the points raised:

**Tabs**

This is also used in the "Install" page*, and tabs are pretty standard when you need to offer different options for e.g. OSs or languages (all the Getting Started guides I used as reference use them). IMO toggles aren't as natural a representation for this and could make the page look "crowded". Looking ahead, tabs can be useful in a lot more places, like guides or if we eventually support UDFs in different languages.

Adding a tab for Cloud is a good idea! Would we still keep the getting started guide under "Cloud" if we added it? I also have to say I don't like the huge "Try the demo in your browser" button, but don't have a better suggestion for where to put it or how to include the Katacoda option. 

*I'd even use them for the regular installation instructions, but then nothing would be listed in the table of contents and it could potentially make it less intuitive.

**Joins & Temporal Filters**

The main reason behind the toggles is that it's quite difficult to get a working `JOIN` example based on the PubNub source without fabricating some reference data, so the example was taking a lot of space. I'm not entirely sure whether these belong in a Getting Started guide, to be honest. On one hand, this should be a "from zero to your first materialized view" quickstart; on the other, joins are a differentiating feature in MZ (be it because some tools don't support them at all or because they're somewhat limited in others) and a lot of users ask about time-windows.

I like the idea of having a "What's next?" section (something like what e.g. [Planet Scale](https://docs.planetscale.com/tutorial/connect-any-application#what's-next?) and [Cockroach](https://www.cockroachlabs.com/docs/cockroachcloud/quickstart#learn-more) have) to guide users that want to continue exploring. In our case, pointing to the "SQL Commands" page might be a shot in the foot, since it's pretty hard to parse in its current state. Do you have other suggestions on how to round things up?

**Comment by awang** [comment]
re: Katacoda -- I'm not wedded to it!  Long-term I'd really like to have this built into the cloud product.  A small annoyance with Katacoda is that we don't have unit tests, so it tends to break periodically for weeks before we notice.  A bigger annoyance is that we don't have good analytics, so I rely on user reports + side-by-side sessions

**Comment by morsapaes** [comment]
@benesch, can you also have a look and approve this PR? For the build checks, the failing links are all within the new tabs template, so it might be related but I can't really pinpoint what's causing them to not pass the check.

**Comment by morsapaes** [comment]
The failed checks seem to be related to a failure in resolving partial links inside tabs. Solved it by passing a [`ref`](https://gohugo.io/content-management/cross-references/) to generate the links. 

Thanks to @benesch for pointing out how to run the `htmltest` checks locally, it made things a lot easier to debug! 🙇‍♀️
